tests/test_ext_javadoctest.py

```diff
@@ -0,0 +1,15 @@
+import pytest
+
+
+@pytest.mark.sphinx('javadoctest', testroot='ext-javadoctest')
+def test_build(app, status, warning):
+    app.builder.build_all()
+    if app.statuscode != 0:
```

Comment:
Testing should be done via assertion and not by raising assertions. Use: 

`assert app.statuscode == 0`

It would be good to test the output as well (like, what is *really* the output, line per line, if there are any). Just ensuring that `app.statuscode` does not fail is insufficient I think.